engine/resources/file.go

```diff
@@ -137,9 +137,19 @@ func (obj *FileRes) isDir() bool {
 // the case where the mode is not specified. The caller should check obj.Mode is
 // not empty.
 func (obj *FileRes) mode() (os.FileMode, error) {
-	m, err := strconv.ParseInt(obj.Mode, 8, 32)
+	n, err := strconv.ParseInt(obj.Mode, 8, 32)
+	m := os.FileMode(n)
 	if err != nil {
-		return os.FileMode(0), errwrap.Wrapf(err, "mode should be an octal number (%s)", obj.Mode)
+		// Try parsing symbolic
+		modes := strings.Split(obj.Mode, ",")
```

Comment:
As per @dbakong's comment, I think we should track duplicates so that we don't allow `u=rw,u=r` or `u+rw,u-r` sorts of things. As a result of adding that logic, perhaps this should be in a separate helper function?

Reply:
My initial interpretation of symbolic mode is that it should be idempotent and it shouldn't matter if you gave `u=x,u=r,u=rw` you would just get `u=rw`. Thinking about it further that may allow for some confusing inputs. Should repeated subjects (ex uug=rw) be an error as well?

Reply:
Honestly, unless the `chmod` "standard" allows such weird things (please have a look) then we shouldn't allow them at all. We want to be less permissive on what we accept so that accidentally incorrect input here causes an error instead of silently allowing wrong behaviour.

As a slight warning: I was planning to write this patch (but you're welcome to include it if you like) but while it's great that you support the `+` and `-`, I was thinking of having those be enabled by a bool const. That way, we can leave support for those off by default. My reasoning: I'm contemplating whether they're really necessary here or not. But I like having the possibility to flip it on if we find a good reason. Cheers!